models/git/lfs.go

```diff
@@ -319,3 +333,36 @@ func GetRepoLFSSize(ctx context.Context, repoID int64) (int64, error) {
 	}
 	return lfsSize, nil
 }
+
+// IterateLFSMetaObjectsForRepo provides a iterator for LFSMetaObjects per Repo
+func IterateLFSMetaObjectsForRepo(ctx context.Context, repoID int64, f func(context.Context, *LFSMetaObject, int64) error) error {
+	var start int
+	batchSize := setting.Database.IterateBufferSize
+	sess := db.GetEngine(ctx)
+	type CountLFSMetaObject struct {
+		Count int64
+		LFSMetaObject
+	}
+
+	for {
+		beans := make([]*CountLFSMetaObject, 0, batchSize)
+		// SELECT `lfs_meta_object`.*, COUNT(`l1`.id) as `count` FROM lfs_meta_object INNER JOIN lfs_meta_object AS l1 ON l1.oid = lfs_meta_object.oid WHERE lfs_meta_object.repository_id = ? GROUP BY lfs_meta_object.id
+		sess.Select("`lfs_meta_object`.*, COUNT(`l1`.oid) AS `count`").
+			Join("INNER", "`lfs_meta_object` AS l1", "`lfs_meta_object`.oid = `l1`.oid").
+			Where("`lfs_meta_object`.repository_id = ? AND `lfs_meta_object`.created_unix < ?", repoID, time.Now().Add(-24*7*time.Hour)).
```

Comment:
Why `time.Now().Add(-24*7*time.Hour)`?

Reply:
The order of git lfs upload and git object upload is not guaranteed. It's possible to imagine a situation whereby the an LFS object is uploaded but the git branch is not uploaded immediately, or there are some rapid changes in new branches that might lead to lfs objects becoming temporarily unassociated with the git objects. 

This delay says only garbage collect LFS objects that have been around for a week or more. I think a week is potentially excessive but it should definitely be enough that any unassociated LFS object is genuinely unassociated.

Reply:
I see, you are very thoughtful, and that will be nice to add this clear explanation to code comments.

However, the function is to provide an iterator for LFSMetaObjects, so there should be a parameter to specify a condition of `created_unix`. You know, maybe one day I'll use `IterateLFSMetaObjectsForRepo` to do something else, like counting them.

Reply:
Wait, I think there's a little problem, time is passing, so the result of `time.Now().Add(-24*7*time.Hour)` for each loop is different, and the query results could be different with a small probability.

Reply:
oh yeah!

Reply:
I think that should be handled now.